components/dashboard/src/Menu.tsx

```diff
@@ -146,14 +145,19 @@ export default function Menu() {
     useEffect(() => {
         const { server } = getGitpodService();
         Promise.all([
-            server.getShowPaymentUI().then((v) => () => setShowPaymentUI(v)),
             server.getClientRegion().then((v) => () => {
                 // @ts-ignore
                 setCurrency(countries[v]?.currency === "EUR" ? "EUR" : "USD");
             }),
             server.isStudent().then((v) => () => setIsStudent(v)),
             server.isChargebeeCustomer().then((v) => () => setIsChargebeeCustomer(v)),
         ]).then((setters) => setters.forEach((s) => s()));
+
+        // Refresh billing mode
+        refreshUserBillingMode();
+        if (team) {
+            server.getBillingModeForTeam(team.id).then(setTeamBillingMode);
+        }
```

Comment:
⚠️ This should probably be run again every time the `team` changes (e.g. if you're viewing one team which has this off, then move to a different team -- currently the effect you wrote this in runs only once when the `Menu` component gets mounted).

A separate, team-dependent effect could look like this:

```ts
useEffect(() => {
    if (team) {
        server.get // ...
    }
}, [team]);
```

Reply:
Addressed in https://github.com/gitpod-io/gitpod/pull/11814/commits/0006b8d855ca60773cf3fd1b2b9601e5f407fc32